### PR TITLE
Change Stackdriver stats exporter to use metrics util.

### PR DESCRIPTION
**Pull request by bogdandrutu**
Split metric descriptor creation into a separate MetricExporter in the pipeline. Enhance tests.

EDIT: updates https://github.com/census-instrumentation/opencensus-java/issues/1657.

### CODE REVIEW

File: exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/CreateMetricDescriptorExporter.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+import io.opencensus.trace.Tracing;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class CreateMetricDescriptorExporter extends MetricExporter {
```
Comment (songy23) [inline exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/CreateMetricDescriptorExporter.java]: nit: this class doesn't need to be public.

File: exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/Util.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.exporter.stats.stackdriver;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+
+final class Util {
```
Comment (songy23) [inline exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/Util.java]: These util methods can be under `StackdriverExportUtils`.

File: exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/CreateMetricDescriptorExporter.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+      Logger.getLogger(CreateMetricDescriptorExporter.class.getName());
+
+  private final String projectId;
+  private final ProjectName projectName;
+  private final MetricServiceClient metricServiceClient;
+  private final String domain;
+  private final String displayNamePrefix;
+  private final Map<String, io.opencensus.metrics.export.MetricDescriptor>
+      registeredMetricDescriptors = new LinkedHashMap<>();
+  private final MetricExporter nextExporter;
```
Comment (songy23) [inline exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/CreateMetricDescriptorExporter.java]: nit: the type of `nextExporter` can be restricted to `CreateTimeSeriesExporter`.

Comment (bogdandrutu) [reply exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/CreateMetricDescriptorExporter.java]: I use a different MetricExporter in tests. Also this class is internal.

File: exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/CreateMetricDescriptorExporter.java (hunk 1 above)
Comment (bogdandrutu) [reply exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/CreateMetricDescriptorExporter.java]: done.

File: exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/Util.java (hunk 2 above)
Comment (bogdandrutu) [reply exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/Util.java]: done.

**Pull request merged**